chainer/backends/cuda.py

```diff
@@ -207,6 +207,11 @@ class GpuDevice(_backend.Device):
     def __init__(self, device):
         check_cuda_available()
         assert isinstance(device, Device)
+        num_gpus = cupy.cuda.runtime.getDeviceCount()
+        if not 0 <= device.id < num_gpus:
+            raise ValueError(
+                'CUDA device ID must be lower '
+                'than max gpus:{} id: {}'.format(num_gpus, device.id))
```

Comment:
I think this representation is difficult to interpret.
How about something like this:
"CUDA device ID must be lower than max GPUs. (actual: {deivce.id} > max: {num_gpus})"